File: selection/_060/Main.java

import java.io.PrintWriter;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();
        long[][] dist = new long[n][n];
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if (i == j) {
                    continue;
                }
                dist[i][j] = 1L << 60;
            }
        }
        for (int i = 0; i < m; i++) {
            int s = sc.nextInt();
            int t = sc.nextInt();
            long d = sc.nextLong();
            dist[s][t] = d;
        }
        sc.close();

        for (int k = 0; k < n; k++) {
            for (int i = 0; i < n; i++) {
                for (int j = 0; j < n; j++) {
                    if (dist[i][k] == 1L << 60 || dist[k][j] == 1L << 60) {
                        continue;
                    }
                    dist[i][j] = Math.min(dist[i][k] + dist[k][j], dist[i][j]);
                }
            }
        }

        for (int i = 0; i < n; i++) {
            if (dist[i][i] < 0) {
                System.out.println("NEGATIVE CYCLE");
                return;
            }
        }

        PrintWriter out = new PrintWriter(System.out);
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if (dist[i][j] == 1L << 60) {
                    out.print("INF");
                } else {
                    out.print(dist[i][j]);
                }

                if (j == n - 1) {
                    out.println();

                } else {
                    out.print(" ");
                }
            }
        }

        out.flush();
    }
}

File: selection/_060/main.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
	"strconv"
)

const INF = int(1e16)

var sc = bufio.NewScanner(os.Stdin)

func main() {
	sc.Split(bufio.ScanWords)

	v, e := scanInt(), scanInt()
	g := make([][]int, v)
	for i := 0; i < v; i++ {
		g[i] = make([]int, v)
		for j := 0; j < v; j++ {
			if i == j {
				continue
			}

			g[i][j] = INF
		}
	}
	for i := 0; i < e; i++ {
		s, t, d := scanInt(), scanInt(), scanInt()
		g[s][t] = d
	}

	for k := 0; k < v; k++ {
		for i := 0; i < v; i++ {
			for j := 0; j < v; j++ {
				if g[i][k] == INF || g[k][j] == INF {
					continue
				}

				g[i][j] = min(g[i][j], g[i][k]+g[k][j])
				if i == j && g[i][j] != 0 {
					fmt.Println("NEGATIVE CYCLE")
					return
				}
			}
		}
	}

	for i := 0; i < v; i++ {
		for j := 0; j < v; j++ {
			if g[i][j] == INF {
				if j == v-1 {
					fmt.Print("INF")
				} else {
					fmt.Print("INF", " ")
				}
			} else {
				if j == v-1 {
					fmt.Print(g[i][j])
				} else {
					fmt.Print(g[i][j], " ")
				}
			}
		}
		fmt.Println()
	}
}

func scanInt() int {
	sc.Scan()

	res, err := strconv.Atoi(sc.Text())
	if err != nil {
		panic(err)
	}

	return res
}

func scanStr() string {
	sc.Scan()

	return sc.Text()
}

// a[i] >= v を満たす最小のインデックスを取得する
func lowerBound(a []int, v int) int {
	ng, ok := -1, len(a)
	for ok-ng > 1 {
		mid := (ok + ng) / 2
		if a[mid] >= v {
			ok = mid
		} else {
			ng = mid
		}
	}

	return ok
}

// a[i] > v を満たす最小のインデックスを取得する
func upperBound(a []int, v int) int {
	ng, ok := -1, len(a)
	for ok-ng > 1 {
		mid := (ok + ng) / 2
		if a[mid] > v {
			ok = mid
		} else {
			ng = mid
		}
	}

	return ok
}

func min(n ...int) int {
	res := n[0]

	for _, v := range n {
		res = int(math.Min(float64(res), float64(v)))
	}

	return res
}

func max(n ...int) int {
	res := n[0]

	for _, v := range n {
		res = int(math.Max(float64(res), float64(v)))
	}

	return res
}

func abs(a int) int {
	return int(math.Abs(float64(a)))
}

func sqrt(x int) int {
	return int(math.Sqrt(float64(x)))
}
